### PR TITLE
Implement (future) item trades

### PR DESCRIPTION
**Pull request by TeeSeal**
Please review, senpais.

### CODE REVIEW

**Comment by iCrawl** [comment]
Can you like move the convert somewhere else? I dunno, some class with items or something? I wanna avoid code duplication.

**Comment by TeeSeal** [comment]
>Can you like move the convert somewhere else?

In `ItemGroup` is fine?

**Comment by iCrawl** [comment]
Anywhere really, just avoid duplication like that.

**Comment by TeeSeal** [comment]
Aight

File: commands/item/give.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const { Command } = require('discord.js-commando');
+
+const Inventory = require('../../currency/Inventory');
+const ItemGroup = require('../../currency/ItemGroup');
+
+module.exports = class ItemGiveCommand extends Command {
+	constructor(client) {
+		super(client, {
+			name: 'give',
+			aliases: [
```
Comment (iCrawl) [inline commands/item/give.js]: Only newline aliases if they would beak 120 max-len, please.

File: commands/item/trade.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+const Currency = require('../../currency/Currency');
+const Inventory = require('../../currency/Inventory');
+const ItemGroup = require('../../currency/ItemGroup');
+
+module.exports = class ItemTradeCommand extends Command {
+	constructor(client) {
+		super(client, {
+			name: 'item-trade',
+			aliases: [
```
Comment (iCrawl) [inline commands/item/trade.js]: Ditto.

File: commands/item/trade.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			args: [
+				{
+					key: 'member',
+					prompt: 'what user would you like to give your item(s)?\n',
+					type: 'member'
+				},
+				{
+					key: 'offerAmount',
+					label: 'amount of items to give',
+					prompt: 'how many would you like to offer?\n',
```
Comment (iCrawl) [inline commands/item/trade.js]: how many items would you like to trade?\n

File: commands/item/trade.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+					type: 'integer',
+					min: 1
+				},
+				{
+					key: 'offerItem',
+					prompt: 'what item would you like to offer?\n',
```
Comment (iCrawl) [inline commands/item/trade.js]: what item would you like to trade?\n

File: commands/item/trade.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+					type: 'string'
+				},
+				{
+					key: 'receiveAmount',
+					label: 'amount of items to receive',
+					prompt: 'how many would you like to receive?\n',
```
Comment (iCrawl) [inline commands/item/trade.js]: how many items would you like to receive?\n

File: commands/item/trade.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+				{
+					key: 'receiveItem',
+					prompt: 'what item would you like to receive?\n',
+					type: 'string'
+				}
+			]
+		});
+	}
+
+	async run(msg, args) {
```
Comment (iCrawl) [inline commands/item/trade.js]: All good, but maybe think about switching `amount` with `item`.
This way users can select the item before the amount.
Also you maybe want custom validation on those, to check if the user even has that item, or if he has enough of that item.

File: commands/item/trade.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+		const receiveItemBalance = receiveInv.content[receiveItem] ? receiveInv.content[receiveItem].amount : 0;
+
+		if (!offerItem && (offerAmount > offerBalance)) return msg.reply(`you have ${Currency.convert(offerBalance)}`);
+		if (!receiveItem && (receiveAmount > receiveBalance)) return msg.reply(`${user.displayName} has ${Currency.convert(receiveBalance)}`);
+		if (offerAmount > offerItemBalance) return msg.reply(`,you have ${offerItemBalance} ${offerItem}s.`);
+		if (receiveAmount > receiveItemBalance) return msg.reply(`,${user.displayName} has ${receiveItemBalance} ${receiveItem}s.`);
+
+		const embed = {
+			title: `${msg.member.displayName} < -- > ${user.displayName}`,
+			description: 'Type `accept` within the next 30 seconds to accept this offer.',
```
Comment (iCrawl) [inline commands/item/trade.js]: Type `accept` within the next 30 seconds to accept the tade.

File: commands/item/trade.js (hunk 6 above)
Comment (TeeSeal) [inline commands/item/trade.js]: Sure can do. I tried to formulate it as naturally as possible. As in:
I want to `trade 5 cards` *for* `300 donuts`

**Pull request merged**